interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -1635,7 +1639,14 @@ namespace {
       = const_cast<SrcMgr::ContentCache&>(MainFileFI.getContentCache());
     if (!Buffer)
       Buffer = llvm::MemoryBuffer::getMemBuffer("/*CLING DEFAULT MEMBUF*/;\n");
-    MainFileCC.setBuffer(std::move(Buffer));
+    if (AutoComplete) {
+      // Adapted from upstream clang/lib/Interpreter/Interpreter.cpp
+      // FIXME: Merge with CompilerInstance::ExecuteAction.
+      llvm::MemoryBuffer* MB = Buffer.release();
+      CI->getPreprocessorOpts().addRemappedFile(Filename, MB);
+    } else {
+      MainFileCC.setBuffer(std::move(Buffer));
+    }
```

Comment:
Can the code path using `addRemappedFile` potentially be used for both code paths?